### PR TITLE
Add pgAdmin to docker compose for nice local admin UI

### PR DESCRIPTION
**Pull request by jamesgorrie**
# Description

- adds [pgAdmin](https://www.pgadmin.org/) to docker compose to give us a nice UI into the DB
  Screenshot 2024-11-26 at 17 38 36
- couple of [yamllint](https://yamllint.readthedocs.io/en/stable/) fixes
  
## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

This is a local change so can be tested by running `make run` and visiting http://localhost:8000

I'll be making some follow up changes, so I'll add some docs while there.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -32,6 +31,49 @@ services:
...
+    environment:
+      # This deavctivates the login screen
+      PGADMIN_CONFIG_SERVER_MODE: "False"
+      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: "False"
+    volumes:
+      - pgadmin-data:/var/lib/pgadmin
+    # This is just nice to know as if it fails, it fails quietly
+    healthcheck:
+      test: [CMD, wget, -O, "-", http://localhost:80/misc/ping]
+configs:
```
Comment (katybaulch) [inline docker-compose.yml:58]: I'd love some more info on this `configs` section as I haven't come across it before! Wha's the benefit of this over extending the out of the box pgadmin4 image and configuring inside that extended image?

File: docker-compose.yml [hunk 2]
```diff
@@ -32,6 +31,49 @@ services:
...
+  pgpass:
+    content: "admin_backend_db:5432:*:${POSTGRES_USER}:${POSTGRES_PASSWORD}"
+  servers.json:
+    content: |
+      {
+        "Servers": {
+          "1": {
+            "Group": "Servers",
+            "Name": "Navigator Admin Backend",
+            "Host": "admin_backend_db",
```
Comment (katybaulch) [inline docker-compose.yml:68]: Should we be using `PGHOST` for this? What if someone changes the value in the .env file?

File: docker-compose.yml [hunk 3]
```diff
@@ -32,6 +31,49 @@ services:
...
+      PGADMIN_CONFIG_SERVER_MODE: "False"
+      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: "False"
+    volumes:
+      - pgadmin-data:/var/lib/pgadmin
+    # This is just nice to know as if it fails, it fails quietly
+    healthcheck:
+      test: [CMD, wget, -O, "-", http://localhost:80/misc/ping]
+configs:
+  pgpass:
+    content: "admin_backend_db:5432:*:${POSTGRES_USER}:${POSTGRES_PASSWORD}"
```
Comment (katybaulch) [inline docker-compose.yml:60]: Should we be using `PGHOST` for this? What if someone changes the value in the .env file?

File: docker-compose.yml (hunk 1 above)
Comment (jamesgorrie) [reply docker-compose.yml:58]: When you say extending the pg4admin image - what do you mean? Creating a new `Containerfile` for it?

My thinking on this was that pgAdmin makes these options configurable via files that exist on the filesystem at startup. A way that's recommended by Docker is to use [configs](https://docs.docker.com/reference/compose-file/services/#configs) ([here's the top level element](https://docs.docker.com/reference/compose-file/configs/)).

I was you could specify it directly on the specific `service` - but their API isn't shaped that way.

Not sure I've answered the question?

File: docker-compose.yml (hunk 3 above)
Comment (jamesgorrie) [reply docker-compose.yml:60]: I thought hardcoding this to what's in the `Dockerfile` made sense as `localhost` wouldn't work here as the networking is happening between the services specified in this file.

File: docker-compose.yml (hunk 2 above)
Comment (jamesgorrie) [reply docker-compose.yml:68]: I'll keep the discussion running in the above comment.

File: docker-compose.yml (hunk 1 above)
Comment (katybaulch) [reply docker-compose.yml:58]: Thanks for sharing - I hadn't come across this before! Good to know :)

**Review by katybaulch — CHANGES_REQUESTED** [review]
Code quality checks are failing but otherwise happy -- following up from our chat just now in the office I think we might benefit from starting to write our own dockerfiles and store them in ECR -- as I imagine this might be reusable between repos :) Nice work!

**Review by katybaulch — APPROVED** [review]
lgtm, we probably want to fix the conflicting lint rules going forward though rather than ignoring

File: docker-compose.yml [hunk 4]
```diff
@@ -46,5 +46,56 @@ services:
...
+      PGADMIN_CONFIG_SERVER_MODE: "False"
+      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: "False"
+    volumes:
+      - pgadmin-data:/var/lib/pgadmin
+    # This is just nice to know as if it fails, it fails quietly
+    healthcheck:
+      # there are conflicting trunk errors here. Either
+      # - having `:` in the URL (fails in Github actions)
+      # - redundantly quoting the URL (fails locally)
+      # trunk-ignore(yamllint/quoted-strings)
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply docker-compose.yml:75]: Yea, I would have tried a bit more, but it's blocking this branch which was getting conflicts.

The annoying thing about this conflict is that everything seems to pass locally, but I might need to try understand the different lifecycles of trunk, which is where my knowledge gap is.

**Pull request merged**